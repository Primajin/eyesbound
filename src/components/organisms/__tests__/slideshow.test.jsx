import React from 'react';
import {create} from 'react-test-renderer';

import Slideshow from '../slideshow.jsx';

describe('Slideshow', () => {
	it('renders correctly with a list of zero', () => {
		const component = create(<Slideshow images={[]}/>);
		expect(component).toMatchSnapshot();
	});

	it('renders correctly with a list of one', () => {
		const picture = {coordinates: {latitude: 52.123, longitude: 13.123}, image: {}, title: 'title1'};
		const prismicPicture = {data: picture, id: 'id-foo', uid: 'uid-bar'};
		const prismicPictureNode = {node: prismicPicture};
		const images = [prismicPictureNode];
		const component = create(<Slideshow isFullscreen images={images}/>);
		expect(component).toMatchSnapshot();
	});

	it('renders correctly with a list of more', () => {
		const picture1 = {coordinates: {latitude: 52.456, longitude: 13.456}, image: {}, title: 'title1'};
		const picture2 = {coordinates: {latitude: 52.789, longitude: 13.798}, image: {}, title: 'title2'};
		const prismicPicture1 = {data: picture1, id: 'id-1', uid: 'uid-1'};
		const prismicPicture2 = {data: picture2, id: 'id-2', uid: 'uid-2'};
		const prismicPictureNode1 = {node: prismicPicture1};
		const prismicPictureNode2 = {node: prismicPicture2};
		const images = [prismicPictureNode1, prismicPictureNode2];
		const component = create(<Slideshow isFullscreen images={images}/>);
		expect(component).toMatchSnapshot();
	});
});
