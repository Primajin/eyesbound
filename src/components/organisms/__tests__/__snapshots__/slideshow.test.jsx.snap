// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slideshow renders correctly with list of more 1`] = `
<div
  className="slide-container"
>
  <div
    aria-roledescription="carousel"
    dir="ltr"
  >
    <div
      className="react-slideshow-container"
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      onMouseUp={[Function]}
      onTouchCancel={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
    >
      <button
        aria-label="Previous Slide"
        className="isFullscreen nav "
        css={
          Object {
            "map": undefined,
            "name": "13ys6af",
            "next": undefined,
            "styles": "
	background: var(--foreground);
	color: var(--background);
	height: 180px;
	opacity: 1;
	position: fixed;
	text-align: center;
	top: 50%;
	transform: translateY(-50%);
	transition: background-color .333s, color .333s, opacity .333s;
	width: 40px;
	z-index: 5;

	@media (min-width: 768px) {
		width: 20px;
	}

	&:hover,
	&:focus {
		background: var(--background);
		color: var(--foreground);
		outline: 0;
	}

	&.isFullscreen {
		opacity: 0.65;
	}

	&[data-type=\\"next\\"] {
		right: 0;
	}
",
            "toString": [Function],
          }
        }
        data-type="prev"
        disabled={false}
        onClick={[Function]}
        tabIndex="10"
        type="button"
      >
        «
      </button>
      <div
        className="react-slideshow-fade-wrapper "
      >
        <div
          className="react-slideshow-fade-images-wrap"
        >
          <div
            aria-hidden="false"
            aria-roledescription="slide"
            data-index={0}
            style={
              Object {
                "opacity": "1",
                "zIndex": "1",
              }
            }
          >
            <a
              href="/picture/uid-1"
              title="title1"
            >
              <figure>
                <picture>
                  <img
                    alt="title1"
                    css={
                      Object {
                        "map": undefined,
                        "name": "724za4",
                        "next": undefined,
                        "styles": "
	height: calc(100vh - 50px);
	width: 100vw;
",
                        "toString": [Function],
                      }
                    }
                    image={
                      Object {
                        "layout": "FULL_WIDTH",
                      }
                    }
                  />
                </picture>
                <figcaption
                  css={
                    Object {
                      "map": undefined,
                      "name": "t2dtan",
                      "next": undefined,
                      "styles": "
	display: none;
",
                      "toString": [Function],
                    }
                  }
                >
                  title1
                </figcaption>
              </figure>
            </a>
          </div>
          <div
            aria-hidden="true"
            aria-roledescription="slide"
            data-index={1}
            style={
              Object {
                "opacity": "0",
                "zIndex": "0",
              }
            }
          >
            <a
              href="/picture/uid-2"
              title="title2"
            >
              <figure>
                <picture>
                  <img
                    alt="title2"
                    css={
                      Object {
                        "map": undefined,
                        "name": "724za4",
                        "next": undefined,
                        "styles": "
	height: calc(100vh - 50px);
	width: 100vw;
",
                        "toString": [Function],
                      }
                    }
                    image={
                      Object {
                        "layout": "FULL_WIDTH",
                      }
                    }
                  />
                </picture>
                <figcaption
                  css={
                    Object {
                      "map": undefined,
                      "name": "t2dtan",
                      "next": undefined,
                      "styles": "
	display: none;
",
                      "toString": [Function],
                    }
                  }
                >
                  title2
                </figcaption>
              </figure>
            </a>
          </div>
        </div>
      </div>
      <button
        aria-label="Next Slide"
        className="isFullscreen nav "
        css={
          Object {
            "map": undefined,
            "name": "13ys6af",
            "next": undefined,
            "styles": "
	background: var(--foreground);
	color: var(--background);
	height: 180px;
	opacity: 1;
	position: fixed;
	text-align: center;
	top: 50%;
	transform: translateY(-50%);
	transition: background-color .333s, color .333s, opacity .333s;
	width: 40px;
	z-index: 5;

	@media (min-width: 768px) {
		width: 20px;
	}

	&:hover,
	&:focus {
		background: var(--background);
		color: var(--foreground);
		outline: 0;
	}

	&.isFullscreen {
		opacity: 0.65;
	}

	&[data-type=\\"next\\"] {
		right: 0;
	}
",
            "toString": [Function],
          }
        }
        data-type="next"
        disabled={false}
        onClick={[Function]}
        tabIndex="11"
        type="button"
      >
        »
      </button>
    </div>
    <ul
      className="indicators"
    >
      <li
        aria-label="Go to slide 1"
        className="indicator active"
        data-key={0}
        onClick={[Function]}
        title="title1"
      >
        <figure>
          <picture>
            <img
              alt="title1"
              css={
                Object {
                  "map": undefined,
                  "name": "1hgrnzr",
                  "next": undefined,
                  "styles": "
	height: 31px;
	width: 100%;
",
                  "toString": [Function],
                }
              }
              height={31}
              image={
                Object {
                  "layout": "CONSTRAINED",
                }
              }
              width={46}
            />
          </picture>
          <figcaption
            css={
              Object {
                "map": undefined,
                "name": "1052cht",
                "next": undefined,
                "styles": "
	bottom: 15px;
	color: var(--background);
	display: none;
	font-weight: 700;
	opacity: 0;
	position: absolute;
	right: 20px;
	text-align: right;
	text-transform: uppercase;
	transition: opacity .333s;

	@media (min-width: 768px) {
		display: block;
	}

	.active & {
		opacity: 1;
	}
",
                "toString": [Function],
              }
            }
          >
            title1
          </figcaption>
        </figure>
      </li>
      <li
        aria-label="Go to slide 2"
        className="indicator "
        data-key={1}
        onClick={[Function]}
        title="title2"
      >
        <figure>
          <picture>
            <img
              alt="title2"
              css={
                Object {
                  "map": undefined,
                  "name": "1hgrnzr",
                  "next": undefined,
                  "styles": "
	height: 31px;
	width: 100%;
",
                  "toString": [Function],
                }
              }
              height={31}
              image={
                Object {
                  "layout": "CONSTRAINED",
                }
              }
              width={46}
            />
          </picture>
          <figcaption
            css={
              Object {
                "map": undefined,
                "name": "1052cht",
                "next": undefined,
                "styles": "
	bottom: 15px;
	color: var(--background);
	display: none;
	font-weight: 700;
	opacity: 0;
	position: absolute;
	right: 20px;
	text-align: right;
	text-transform: uppercase;
	transition: opacity .333s;

	@media (min-width: 768px) {
		display: block;
	}

	.active & {
		opacity: 1;
	}
",
                "toString": [Function],
              }
            }
          >
            title2
          </figcaption>
        </figure>
      </li>
    </ul>
  </div>
</div>
`;

exports[`Slideshow renders correctly with list of one 1`] = `
<div
  className="slide-container"
>
  <div
    aria-roledescription="carousel"
    dir="ltr"
  >
    <div
      className="react-slideshow-container"
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      onMouseUp={[Function]}
      onTouchCancel={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
    >
      <button
        aria-label="Previous Slide"
        className="isFullscreen nav "
        css={
          Object {
            "map": undefined,
            "name": "13ys6af",
            "next": undefined,
            "styles": "
	background: var(--foreground);
	color: var(--background);
	height: 180px;
	opacity: 1;
	position: fixed;
	text-align: center;
	top: 50%;
	transform: translateY(-50%);
	transition: background-color .333s, color .333s, opacity .333s;
	width: 40px;
	z-index: 5;

	@media (min-width: 768px) {
		width: 20px;
	}

	&:hover,
	&:focus {
		background: var(--background);
		color: var(--foreground);
		outline: 0;
	}

	&.isFullscreen {
		opacity: 0.65;
	}

	&[data-type=\\"next\\"] {
		right: 0;
	}
",
            "toString": [Function],
          }
        }
        data-type="prev"
        disabled={false}
        onClick={[Function]}
        tabIndex="10"
        type="button"
      >
        «
      </button>
      <div
        className="react-slideshow-fade-wrapper "
      >
        <div
          className="react-slideshow-fade-images-wrap"
        >
          <div
            aria-hidden="false"
            aria-roledescription="slide"
            data-index={0}
            style={
              Object {
                "opacity": "1",
                "zIndex": "1",
              }
            }
          >
            <a
              href="/picture/uid-bar"
              title="title1"
            >
              <figure>
                <picture>
                  <img
                    alt="title1"
                    css={
                      Object {
                        "map": undefined,
                        "name": "724za4",
                        "next": undefined,
                        "styles": "
	height: calc(100vh - 50px);
	width: 100vw;
",
                        "toString": [Function],
                      }
                    }
                    image={
                      Object {
                        "layout": "FULL_WIDTH",
                      }
                    }
                  />
                </picture>
                <figcaption
                  css={
                    Object {
                      "map": undefined,
                      "name": "t2dtan",
                      "next": undefined,
                      "styles": "
	display: none;
",
                      "toString": [Function],
                    }
                  }
                >
                  title1
                </figcaption>
              </figure>
            </a>
          </div>
        </div>
      </div>
      <button
        aria-label="Next Slide"
        className="isFullscreen nav "
        css={
          Object {
            "map": undefined,
            "name": "13ys6af",
            "next": undefined,
            "styles": "
	background: var(--foreground);
	color: var(--background);
	height: 180px;
	opacity: 1;
	position: fixed;
	text-align: center;
	top: 50%;
	transform: translateY(-50%);
	transition: background-color .333s, color .333s, opacity .333s;
	width: 40px;
	z-index: 5;

	@media (min-width: 768px) {
		width: 20px;
	}

	&:hover,
	&:focus {
		background: var(--background);
		color: var(--foreground);
		outline: 0;
	}

	&.isFullscreen {
		opacity: 0.65;
	}

	&[data-type=\\"next\\"] {
		right: 0;
	}
",
            "toString": [Function],
          }
        }
        data-type="next"
        disabled={false}
        onClick={[Function]}
        tabIndex="11"
        type="button"
      >
        »
      </button>
    </div>
    <ul
      className="indicators"
    >
      <li
        aria-label="Go to slide 1"
        className="indicator active"
        data-key={0}
        onClick={[Function]}
        title="title1"
      >
        <figure>
          <picture>
            <img
              alt="title1"
              css={
                Object {
                  "map": undefined,
                  "name": "1hgrnzr",
                  "next": undefined,
                  "styles": "
	height: 31px;
	width: 100%;
",
                  "toString": [Function],
                }
              }
              height={31}
              image={
                Object {
                  "layout": "CONSTRAINED",
                }
              }
              width={46}
            />
          </picture>
          <figcaption
            css={
              Object {
                "map": undefined,
                "name": "1052cht",
                "next": undefined,
                "styles": "
	bottom: 15px;
	color: var(--background);
	display: none;
	font-weight: 700;
	opacity: 0;
	position: absolute;
	right: 20px;
	text-align: right;
	text-transform: uppercase;
	transition: opacity .333s;

	@media (min-width: 768px) {
		display: block;
	}

	.active & {
		opacity: 1;
	}
",
                "toString": [Function],
              }
            }
          >
            title1
          </figcaption>
        </figure>
      </li>
    </ul>
  </div>
</div>
`;

exports[`Slideshow renders correctly with list of zero 1`] = `
<div
  className="slide-container"
>
  <div
    aria-roledescription="carousel"
    dir="ltr"
  >
    <div
      className="react-slideshow-container"
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      onMouseUp={[Function]}
      onTouchCancel={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
    >
      <button
        aria-label="Previous Slide"
        className=" nav "
        css={
          Object {
            "map": undefined,
            "name": "13ys6af",
            "next": undefined,
            "styles": "
	background: var(--foreground);
	color: var(--background);
	height: 180px;
	opacity: 1;
	position: fixed;
	text-align: center;
	top: 50%;
	transform: translateY(-50%);
	transition: background-color .333s, color .333s, opacity .333s;
	width: 40px;
	z-index: 5;

	@media (min-width: 768px) {
		width: 20px;
	}

	&:hover,
	&:focus {
		background: var(--background);
		color: var(--foreground);
		outline: 0;
	}

	&.isFullscreen {
		opacity: 0.65;
	}

	&[data-type=\\"next\\"] {
		right: 0;
	}
",
            "toString": [Function],
          }
        }
        data-type="prev"
        disabled={false}
        onClick={[Function]}
        tabIndex="10"
        type="button"
      >
        «
      </button>
      <div
        className="react-slideshow-fade-wrapper "
      >
        <div
          className="react-slideshow-fade-images-wrap"
        />
      </div>
      <button
        aria-label="Next Slide"
        className=" nav "
        css={
          Object {
            "map": undefined,
            "name": "13ys6af",
            "next": undefined,
            "styles": "
	background: var(--foreground);
	color: var(--background);
	height: 180px;
	opacity: 1;
	position: fixed;
	text-align: center;
	top: 50%;
	transform: translateY(-50%);
	transition: background-color .333s, color .333s, opacity .333s;
	width: 40px;
	z-index: 5;

	@media (min-width: 768px) {
		width: 20px;
	}

	&:hover,
	&:focus {
		background: var(--background);
		color: var(--foreground);
		outline: 0;
	}

	&.isFullscreen {
		opacity: 0.65;
	}

	&[data-type=\\"next\\"] {
		right: 0;
	}
",
            "toString": [Function],
          }
        }
        data-type="next"
        disabled={false}
        onClick={[Function]}
        tabIndex="11"
        type="button"
      >
        »
      </button>
    </div>
    <ul
      className="indicators"
    />
  </div>
</div>
`;
